src/main/java/tictactoe_game/implementation/WinnerImplementation.java

```diff
@@ -0,0 +1,103 @@
+package tictactoe_game.implementation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import tictactoe_game.model.types.*;
+import tictactoe_game.model.Element;
+import tictactoe_game.model.Grid;
+
+public class WinnerImplementation {
+    public static Boolean is_there_a_winner(Grid grid){
+        var rows = getGridRows(grid);
+        var columns = getGridColumns(grid);
+        var mainDiagonal = getGridMainDiagonal(grid);
+        var secondaryDiagonal = getGridSecondaryDiagonal(grid);
+        List<List<Element>> list = new ArrayList<List<Element>>();
+        
+        list.addAll(rows);
+        list.addAll(columns);
+        list.addAll(mainDiagonal);
+        list.addAll(secondaryDiagonal);
+
+        for (List<Element> elementsList : list) {
+            if (are_the_elements_of_the_list_the_same(elementsList)){
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static Boolean are_the_elements_of_the_list_the_same(List<?> list){
+        if (list.isEmpty()){
```

Comment:
Please, try to avoid multiple returns if they are not serving a specific purpose /e.g. short-circuiting/.